### ISSUE TITLE
Make border utilities hoverable

### ISSUE THREAD

**Issue by spyric** [tailwindcss/tailwindcss#67]
Hello all!

Just tried TailwindCSS and It's perfect, but I think that borders should be able to be combined with `hover:` preffix.

Here is sample. Task make link underline dashed:
`<a href="#" class="no-underline border-b hover:border-b-0">Text link</a>`

Similar approach with manipulating borders are used on https://laravel.com/ nav menu

**Comment by MichaelDeBoey** [comment]
I think we can have most of the utilities be combined with `hover:`, like we have with the responsive ones (`sm`, `md`, `lg` & `xl`)?

**Comment by adamwathan** [comment]
Border *colors* are hoverable which might be enough for your situation.

How does this work for you?

```
<a href="#" class="no-underline border-b hover:border-b-transparent">Text link</a>
```

This is how I usually handle these sorts of things, because otherwise introducing a border when it wasn't there before can shift the content around.

**Comment by spyric** [comment]
@adamwathan Yeah, you're right, It's work exactly that I want.

**Comment by sahin85** [comment]
How can we achieve this " border bottom 2px solid on hover" in Tailwindcss? 


<a href="#" class="m-4 hover:border-b-2 border-blue-600">hover</a>

do we have to add such a code to config file?


cheers 
sahin

**Comment by mooijtech** [comment]
I'm also trying to get a border bottom on hover, this doesn't apply any of the hover classes, why?

`<a class="text-white hover:font-bold hover:border-b hover:border-white mr-12" href="/about">About us</a>`

**Comment by Qwiso** [comment]
ended up here by asking the same question
didn't find a clear answer at first, so here is the solution

tl;dr - modify your configuration and `npm run dev` after
```
// tailwind.config.js
module.exports = {
  // ...
  variants: {
    extend: {
      borderRadius: ['hover'],
    }
  },
}
```

------------

https://tailwindcss.com/docs/hover-focus-and-other-states#hover

it reads as follows

> By default, the hover variant is enabled for the following core plugins:
- backgroundColor
- backgroundOpacity
- borderColor
- borderOpacity
- boxShadow
- gradientColorStops
- opacity
- rotate
- scale
- skew
- textColor
- textDecoration
- textOpacity
- translate

You can control whether hover variants are enabled for a plugin in the variants section of your tailwind.config.js file:
```
// tailwind.config.js
module.exports = {
  // ...
  variants: {
    extend: {
      padding: ['hover'],
    }
  },
}
```

**Comment by Qwiso** [comment]
to @mooijtech and future clicks

`borderRadius` class does not work with the `hover:` state. this is probably intended because you achieve the same functionality 
by doing something like (for a nice menu style) `class="border-b-2 border-double border-transparent hover:border-gray-500 cursor-pointer select-none"`

**Comment by 6LYTH3** [comment]
Per [Tailwindcss default variants reference](https://tailwindcss.com/docs/configuring-variants#default-variants-reference)

we can enable more variants to support the utilities we want by configure in  tailwind.config.js

Example configuration for support
```
<p className='text-gray-500 font-thin hover:border-b hover:border-black'>hover me!</p>
```
Add more pseudo class
```
// tailwind.config.js
variants: {
  extend: {
    borderStyle: ['responsive', 'hover'],
    borderWidth: ['responsive', 'hover'],
    },
},
```

**Comment by Prajapatisantu** [comment]
first set border then set opacity 0 and on hover:set opacity 100, so problem solve on hover set border.
